core-bundle/assets/controllers/sticky-observer-controller.js

```diff
@@ -0,0 +1,25 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    connect() {
+        this.observer = new IntersectionObserver(this.#handleIntersect.bind(this), this.observerOptions);
+        this.observer.observe(this.element);
+    }
+
+    disconnect() {
+        this.observer.disconnect();
+    }
+
+    get observerOptions() {
+        return {
+            threshold: 1.0,
+            rootMargin: '0px 0px -1px 0px',
+        };
+    }
```

Comment:
Just out of curiosity: does it make sense to use `get` here - over just a regular private member variable, since it's a static object you are returning? Isn't that better suited for when you have to return a dynamically computed/accessed value?

Reply:
Unless someone wants to modify it, using a private variable makes more sense, I agree

Reply:
Of course it does not. The original example contained a custom root element and therefore the object wasn't static there. Changed in c9a06588757b9aaa1d5159219594e53776c3767e.

Reply:
We could simplify it even more as there is no need to have it in a private variable (unless for better readability), but w/e.

```js
export default class extends Controller {
    connect() {
        this.observer = new IntersectionObserver(this.#handleIntersect.bind(this), {
            threshold: 1.0,
            rootMargin: '0px 0px -1px 0px',
        });
        this.observer.observe(this.element);
    }

    disconnect() {
        this.observer.disconnect();
    }

    #handleIntersect(entries) {
        for (const entry of entries) {
            this.element.classList.toggle('is-stuck', !entry.isIntersecting);
        }
    }
}

Reply:
nvm, was already done like that :D